source/agora/consensus/EnrollmentManager.d

```diff
@@ -555,7 +555,11 @@ public class EnrollmentManager
             const preimage = this.validator_set.getPreimageAt(key, height);
             // this should not happen. validators which didn't reveal the
             // preimage should not be in the active validator set 'keys'.
-            assert(preimage != PreImageInfo.init);
+            try
+            {
+                assert(preimage != PreImageInfo.init,
+                    format!"No preimage at height %s for validator key %s"(height, key));
+            } catch (Exception) {}
```

Comment:
This ignore the failure. If it can ever happen, it should be handled one way or another.

Reply:
I wanted to keep the function nothrow but format is not. Is there a dlang trick for this?

Reply:
Was updated to not need the try / catch

Reply:
The `try...catch` block is sometimes the workaround.
Does the function have to be `nothrow`?

Reply:
Actually was fixed by using the correct format specifier for height (%d) and for param using height.value.
Then there was no potential throw from the format statement.
Always better to use `nothrow` whenever possible.